lib/cosmos/gui/qt_tool.rb

```diff
@@ -69,6 +69,35 @@ def initialize_actions
       end
     end
 
+    # Creates a default menu action based on the default_dir and iterates
+    # through all the target directories adding the target_sub_dir directory if
+    # it exists. Calls the callback option when the action is triggered.
+    def target_dirs_action(menu, default_dirs, target_sub_dir, callback, status_tip = nil)
```

Comment:
Take a look at this because it was my attempt to add a generic method to get all the target directories into the menu. I added it to the ScriptRunner code base as well.